### ISSUE TITLE
npm_lifecycle_event should be set

### ISSUE THREAD

**Issue by e111077** [stackblitz/webcontainer-core#868]
### Describe the bug

`process.env.npm_lifecycle_event` is `undefined` and should be set.

### Link to the blitz that caused the error

https://stackblitz.com/edit/node-rharbn?file=index.js,package.json

### Steps to reproduce

1. go to link above
2. run `npm run test` in terminal
3. observe the console output and that it is `undefined`

### Expected behavior

`test` should be console logged

### Parity with Local

- [X] I have [run the project in my local machine](https://github.com/stackblitz/webcontainer-core/blob/main/repro.md#try-locally) and I could not reproduce the issue.

### Platform

- OS: macOS, Windows, Linux, npm, node
- Browser: Chrome, Safari, Firefox
- Version: all

### Additional context

This might enable projects like https://github.com/google/wireit or https://github.com/vercel/turborepo to work in web containers.

This is a duplicate of https://github.com/stackblitz/core/issues/1857 Happy to close this if that issue is transferred over

**Comment by jrvidal** [comment]
:wave: Thanks for the report.

We are actually working on supporting full, unmodified NPM, so this is something that should be supported soon.

**Comment by matthewp** [comment]
@jrvidal That's amazing! I assume that would mean the --workspace flag would also work?

**Comment by d3lm** [comment]
Yes, that's right. The `--workspace` flag would work then as well 👍
